stores/moderationSearchHealthcareProfessionalsStore.ts

```diff
@@ -0,0 +1,49 @@
+import { gql } from 'graphql-request'
+import { defineStore } from 'pinia'
+import { type Ref, ref } from 'vue'
+import { gqlClient } from '../utils/graphql.js'
+import type { HealthcareProfessional } from '~/typedefs/gqlTypes'
+
+export const useModerationSearchHealthcareProfessionalsStore = defineStore(
+    'moderationSearchHealthcareProfessionalsStore',
+    () => {
+        const healthcareProfessional: Ref<HealthcareProfessional[]> = ref([])
+        async function getHealthcareProfessionals(id: string) {
+            const healthcareProfessionalsResult = await queryHealthcareProfessionalById(id)
+            healthcareProfessional.value = healthcareProfessionalsResult
+        }
+
+        return {
+            getHealthcareProfessionals,
+            healthcareProfessional
+        }
+    }
+)
+
+async function queryHealthcareProfessionalById(id: string) {
+    try {
+        const healthcareProfessionalId = {
+            healthcareProfessionalId: id
+        }
+        const result = await gqlClient.request<{ healthcareProfessional: HealthcareProfessional[] }>(
+            getHealthcareProfessionalsGqlQuery,
+            healthcareProfessionalId
+        )
+        return result.healthcareProfessional ?? []
```

Comment:
I'd like to know what @ermish thinks, but I think we should return something other than an empty list here. Empty lists are usually used when you expected multiple items to be returned, but there were none. In this case we were only expecting one item.

Reply:
I agree. For getbyid it should return the value. 
If it doesn't exist, usually returning undefined would be appropriate for a single item. 

throwing an exception is also a valid option if we expect to always have a response.